sdks/java/io/mongodb/src/main/java/org/apache/beam/sdk/io/mongodb/MongoDbIO.java

```diff
@@ -0,0 +1,535 @@
+                + "\")," + "$lt:ObjectId(\"" + currentID
+                + "\")}}, " + filter + " ]}";
+            splitSourceList.add(new BoundedMongoDbSource
+                (uri, database, collection, newFilter, numberSplit));
+            newFilter = "{ $and: [ {\"_id\":{$gt:ObjectId(\"" + currentID
+                + "\")}}, " + filter + " ]}";
+          } else {
+            newFilter = "{ $and: [ {\"_id\":{$gt:ObjectId(\"" + lastID
+                + "\")," + "$lte:ObjectId(\"" + currentID + "\")}}, "
+                + filter + " ]}";
+          }
+        } else {
+          if (listIndex == 0) {
+            newFilter = "{\"_id\":{$lte:ObjectId(\"" + currentID + "\")}}";
+          } else if (listIndex == (splitIDList.size() - 1)) {
+            newFilter = "{\"_id\":{$gt:ObjectId(\"" + lastID + "\"),"
+                + "$lt:ObjectId(\"" + currentID + "\")}}";
+            splitSourceList.add(new BoundedMongoDbSource
+                (uri, database, collection, newFilter, numberSplit));
+            newFilter = "{\"_id\":{$gt:ObjectId(\"" + currentID + "\")}}";
+          } else {
+            newFilter = "{\"_id\":{$gt:ObjectId(\"" + lastID + "\"),"
+                + "$lte:ObjectId(\"" + currentID + "\")}}";
+          }
+        }
+        splitSourceList.add(new BoundedMongoDbSource
+            (uri, database, collection, newFilter, numberSplit));
+        lastID = currentID;
+        listIndex++;
+      }
+      return splitSourceList;
+    }
+  }
+
+  private static class BoundedMongoDbReader extends BoundedSource.BoundedReader {
+
+    private String uri;
+    private String database;
+    private String collection;
+    private String filter;
+
+    private MongoClient client;
+    private MongoCursor<Document> cursor;
+    private String current;
+
+    public BoundedMongoDbReader(String uri, String database, String collection, String filter) {
+      this.uri = uri;
+      this.database = database;
+      this.collection = collection;
+      this.filter = filter;
+    }
+
+    @Override
+    public boolean start() {
+      client = new MongoClient(new MongoClientURI(uri));
+
+      MongoDatabase mongoDatabase = client.getDatabase(database);
+
+      MongoCollection<Document> mongoCollection = mongoDatabase.getCollection(collection);
+
+      if (filter == null) {
+        cursor = mongoCollection.find().iterator();
+      } else {
+        Document bson = Document.parse(filter);
+        cursor = mongoCollection.find(bson).iterator();
+      }
+
+      return advance();
+    }
+
+    @Override
+    public boolean advance() {
+
+      if (cursor.hasNext()) {
+        current = cursor.next().toJson();
+        return true;
+      } else {
+        return false;
+      }
+    }
+
+    @Override
+    public BoundedSource getCurrentSource() {
+      return null;
+    }
+
+    @Override
+    public Object getCurrent() {
+      return current;
+    }
+
+    @Override
+    public void close() {
+      cursor.close();
```

Comment:
Seems like if cursor.close() throws an exception, then you won't call client.close(). Is this safe?